gobblin-core/src/main/java/org/apache/gobblin/converter/grok/GrokToJsonConverter.java

```diff
@@ -0,0 +1,218 @@
+  private static final String TYPE_KEY = "type";
+  private static final String NULLABLE = "isNullable";
+
+  public static final String PATTERN = "converter.grok_to_json.pattern";
+  public static final String PATTERNS_FILE = "converter.grok_to_json.patterns";
+  public static final String EMPTY_AS_NULL = "converter.grok_to_json.empty_as_null";
+  public static final String NULL_STRING = "converter.grok_to_json.null_string";
+
+  public static final String DEFAULT_PATTERNS_FILE = "/grok/grok-patterns";
+
+  private String nullString;
+  private Boolean emptyAsNull;
+  private Grok grok;
+
+  @Override
+  public Converter<String, JsonArray, String, JsonObject> init(WorkUnitState workUnit) {
+    super.init(workUnit);
+    String pattern = workUnit.getProp(PATTERN);
+    String patternsFile = workUnit.getProp(PATTERNS_FILE);
+    this.nullString = workUnit.getProp(NULL_STRING);
+    this.emptyAsNull = workUnit.getPropAsBoolean(EMPTY_AS_NULL, true);
+
+    InputStreamReader grokPatterns;
+    try {
+      if (patternsFile == null) {
+         grokPatterns = new InputStreamReader(getClass().getResourceAsStream("/grok/grok-patterns"), "UTF8");
+      }else {
+         grokPatterns = new InputStreamReader(new FileInputStream(patternsFile), "UTF8");
+      }
+      grok = new Grok();
+      grok.addPatternFromReader(grokPatterns);
+      grok.compile(pattern);
+    } catch (GrokException | FileNotFoundException | UnsupportedEncodingException e) {
+      throw new RuntimeException("Error initializing GROK: "+ e);
+    }
+
+    return this;
+  }
+
+  @Override
+  public JsonArray convertSchema(String inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
+    Preconditions.checkNotNull(inputSchema, "inputSchema is required.");
+    return JSON_PARSER.parse(inputSchema).getAsJsonArray();
+  }
+
+  /**
+   * Converts Text (String) to JSON based on a Grok regexp expression.
+   * By default, fields between Text and JSON are mapped by Grok SEMANTIC which is the identifier you give to the piece of text being matched in your Grok expression.
+   *
+   *
+   * e.g:
+   * {@inheritDoc}
+   * @see Converter#convertRecord(Object, Object, WorkUnitState)
+   */
+  @Override
+  public Iterable<JsonObject> convertRecord(JsonArray outputSchema, String inputRecord, WorkUnitState workUnit)
+      throws DataConversionException {
+
+    JsonObject outputRecord = null;
+    outputRecord = createOutput(outputSchema, inputRecord);
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Converted into " + outputRecord);
+    }
+    return new SingleRecordIterable<JsonObject>(outputRecord);
+  }
+
+  @VisibleForTesting
+  JsonObject createOutput(JsonArray outputSchema, String inputRecord)
+      throws DataConversionException {
+    JsonObject outputRecord = new JsonObject();
+
+    Match gm = grok.match(inputRecord);
+    gm.captures();
+    JsonParser parser = new JsonParser();
+    JsonElement capturesJson = parser.parse(gm.toJson());
```

Comment:
Based on how it's used below, we're more interested in `parser.parse(gm.toJson()).getAsJsonObject()`